modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,375 @@
+
+. Create a new project for webhook resources:
++
+----
+$ oc new-project my-webhook-namespace  <1>
+----
+<1> Note that the webhook server may expect a specific name.
+
+. Define RBAC rules for the aggregated API service in a file called `rbac.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: List
+items:
+
+- apiVersion: rbac.authorization.k8s.io/v1  <1>
+  kind: ClusterRoleBinding
+  metadata:
+    name: auth-delegator-my-webhook-namespace
+  roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
+    name: system:auth-delegator
+  subjects:
+  - kind: ServiceAccount
+    namespace: my-webhook-namespace
+    name: server
+
+- apiVersion: rbac.authorization.k8s.io/v1  <2>
+  kind: ClusterRole
+  metadata:
+    annotations:
+    name: system:openshift:online:my-webhook-server
+  rules:
+  - apiGroups:
+    - online.openshift.io
+    resources:
+    - namespacereservations  <3>
+    verbs:
+    - get
+    - list
+    - watch
+
+- apiVersion: rbac.authorization.k8s.io/v1  <4>
+  kind: ClusterRole
+  metadata:
+    name: system:openshift:online:my-webhook-requester
+  rules:
+  - apiGroups:
+    - admission.online.openshift.io
+    resources:
+    - namespacereservations <5>
+    verbs:
+    - create
+
+- apiVersion: rbac.authorization.k8s.io/v1  <6>
+  kind: ClusterRoleBinding
+  metadata:
+    name: my-webhook-server-my-webhook-namespace
+  roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
+    name: system:openshift:online:my-webhook-server
+  subjects:
+  - kind: ServiceAccount
+    namespace: my-webhook-namespace
+    name: server
+
+- apiVersion: rbac.authorization.k8s.io/v1  <7>
+  kind: RoleBinding
+  metadata:
+    namespace: kube-system
+    name: extension-server-authentication-reader-my-webhook-namespace
+  roleRef:
+    kind: Role
+    apiGroup: rbac.authorization.k8s.io
+    name: extension-apiserver-authentication-reader
+  subjects:
+  - kind: ServiceAccount
+    namespace: my-webhook-namespace
+    name: server
+
+- apiVersion: rbac.authorization.k8s.io/v1  <8>
+  kind: ClusterRole
+  metadata:
+    name: my-cluster-role
+  rules:
+  - apiGroups:
+    - admissionregistration.k8s.io
+    resources:
+    - validatingwebhookconfigurations
+    - mutatingwebhookconfigurations
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups:
+    - ""
+    resources:
+    - namespaces
+    verbs:
+    - get
+    - list
+    - watch
+
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: my-cluster-role
+  roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
+    name: my-cluster-role
+  subjects:
+  - kind: ServiceAccount
+    namespace: my-webhook-namespace
+    name: server
+----
+<1> Delegates authentication and authorization to the webhook server API.
+<2> Allows the admission webhook server to access cluster resource.
```

Comment:
Should this be "access cluster resources" plural?

Reply:
Resolved.